code/lib/builder-vite/src/plugins/externals-plugin.test.ts

```diff
@@ -0,0 +1,31 @@
+import { rewriteImport } from './externals-plugin';
+
+const packageName = '@storybook/package';
+const globals = { [packageName]: '_STORYBOOK_PACKAGE_' };
+
+const cases = [
+  {
+    globals,
+    packageName,
+    input: `import { Rain, Jour as Day, Nuit as Night, Sun } from "${packageName}"`,
+    output: `const { Rain, Jour: Day, Nuit: Night, Sun } = ${globals[packageName]}`,
+  },
+  {
+    globals,
+    packageName,
+    input: `import{Rain,Jour as Day,Nuit as Night,Sun}from '${packageName}'`,
+    output: `const {Rain,Jour: Day,Nuit: Night,Sun} = ${globals[packageName]}`,
+  },
+  {
+    globals,
+    packageName,
+    input: `const { Afternoon } = await import('${packageName}')`,
+    output: `const { Afternoon } = ${globals[packageName]}`,
+  },
+];
```

Comment:
We need a test for default imports, and I think we might be handling them incorrectly right now.  Correct me if I'm wrong, but I think we should convert from 

```js
import foo from '@storybook/preview-api';
```
to 
```js
const {default: foo} = __STORYBOOK_MODULE_PREVIEW_API__;
```